.github/workflows/turing.yaml

```diff
@@ -339,9 +339,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        name:
-          - "In-cluster credentials"
-          - "Remote cluster credentials"
+        name: ["In-cluster credentials", "Remote cluster credentials"]
```

Comment:
What do you think about only running the API e2e tests for `In-cluster credentials` and letting the SDK-based e2e for the other one (`Remote cluster credentials`) ? As with writing test case, this would be a way to reduce the space of variable combinations. In this case, I don't see much value in keeping separate API e2e tests for `Remote cluster credentials` when we have the same covered by the SDK tests.

Reply:
Along these lines, the regular e2e tests and the SDK based e2e tests could also be combined into a matrix under the same job using a smarter strategy: https://github.com/vlsi/github-actions-random-matrix

But, at this point, this may be over-optimising and we could consider this later instead, if we see a need for it.

Reply:
I think that works, it does sound tedious and wasteful to be running too configurations for all the possible combinations. 

I've made some changes to the CICD workflows in order to adopt your suggestion (run API e2e tests for `In-cluster credentials` and SDK e2e tests for `Remote cluster credentials`): I've removed the `test-e2e-sdk` job to simplify the workflow in `turing.yaml` and instead added the SDK-specific steps ('Setup Python', 'Install Python dependencies', 'Run End-to-End Test Suite with Turing SDK') to the existing `test-e2e` job. These steps only get triggered when the value for `useSDK` is true; conversely, the API e2e tests only get triggered when this is false. I've also changed the matrix strategy to reflect this:

```yaml
      matrix:
        name: ["In-cluster credentials; API e2e", "Remote cluster credentials; SDK e2e"]
        include:
          - name: "In-cluster credentials; API e2e"
            useInClusterConfig: true
            valuesFile: "turing.values.in-cluster.yaml"
            useSDK: false
          - name: "Remote cluster credentials; SDK e2e"
            useInClusterConfig: false
            valuesFile: "turing.values.remote.yaml"
            useSDK: true
```